source/News.h

```diff
@@ -33,8 +35,8 @@ class News {
 	
 	// Check whether this news item has anything to say.
 	bool IsEmpty() const;
-	// Check if this news item is available on the given planet.
-	bool Matches(const Planet *planet) const;
+	// Check if this news item is available given the player's planet and conditions.
+	bool Matches(const PlayerInfo &player) const;
```

Comment:
Let's actually go one better and pass only the planet and conditions, rather than the player.